### DISCUSSION TITLE
Unable to run StreamPipes for Core Development

### DISCUSSION THREAD

**Discussion by himanshu-pareek** [apache/streampipes#2418] [Q&A] [answered]
### Where does your observation occur?

None

### What do you want to report?

I am trying to setup streampipes. I followed the documentation from this website - https://cwiki.apache.org/confluence/display/STREAMPIPES/Core
I did the following:
1. Cloned the github repository.
2. `git checkout dev`.
3. `cd installer/cli && ./streampipes env --set backend`.
4. `./streampipes up -d`
5. Opened the Poject in IDE (IntelliJ IDEA).
6. There are 2 run configurations:
    A. core
    B. StreamPipesCoreApplicationMinimal
7. I ran `core` configuration.
8. I can see that the application is running on port `8030` with context path '/streampipes-backend'.
9. I opened http://localhost:8030/streampipes-backend/ in the browser.
10. I am getting 401 Unauthorized (visible in network logs) and a Whitelabel error page.
11. I have tried some other paths as well which are listed in `UnauthenticatedInterfaces.java` file. But, I am getting the same error for all of them.

I see the following logs in the console repeating. But, I think they are not relevant to the problem:
```bash
2024-01-08T21:40:15.858+05:30  INFO 84218 --- [pool-3-thread-1] o.a.s.s.SpServiceDiscoveryCore           : Could not find any extensions services, retrying (1/3)
2024-01-08T21:40:16.872+05:30  INFO 84218 --- [pool-3-thread-1] o.a.s.s.SpServiceDiscoveryCore           : Could not find any extensions services, retrying (2/3)
2024-01-08T21:40:17.889+05:30  INFO 84218 --- [pool-3-thread-1] o.a.s.s.SpServiceDiscoveryCore           : Could not find any extensions services, retrying (3/3)
2024-01-08T21:40:18.909+05:30  INFO 84218 --- [pool-3-thread-1] o.a.s.s.SpServiceDiscoveryCore           : No service found
```

Am I missing something or doing something wrong? Thank you.

### Solving the problem

I don't know whether this is the issue or if I am doing something wrong. I would appreciate any help getting started contributing as soon as possible.

### Are you willing to submit a PR?

None

**Comment by tenthe** [comment]
Hi @himanshu-pareek,

thank you for reaching out to us.
It looks good to me the way you've done it.
Can you perhaps have a look at the environment variables that are set in IntelliJ for the core?

Cheers,
Philipp

**Comment by himanshu-pareek** [comment]
HI @tenthe 
These are the environment variables which are set for `core` run configuration in IntelliJ:
<img width="748" alt="Screenshot 2024-01-09 at 4 06 26 PM" src="https://github.com/apache/streampipes/assets/17642442/79622a67-7e1b-481b-bc62-9bebca3f5292">

And the following is the complete run configuration for `core`:
<img width="1006" alt="Screenshot 2024-01-09 at 4 08 14 PM" src="https://github.com/apache/streampipes/assets/17642442/1309c78f-203d-489c-9329-d8a1a8e311c1">

Thanks

**Comment by dominikriemer** [comment]
Hi @himanshu-pareek can you access the UI?
It seems the backend is running. It is normal that you get a 401 when accessing the backend directly without authentication. You can also check the output of the extension service running in Docker to make sure it can connect to the locally running backend.

**Comment by himanshu-pareek** [comment]
Hi @dominikriemer 
Thanks for the help. In the output of the command `./streampipes up -d`, I can see that the URL of local docker is `http://localhost/`. I have opened the URL in the browser. And after some time of loading, I see a login page.
Is that the expected behavior? If yes, then what is the email and password to log in?

Thanks.

**Comment by dominikriemer** [comment, marked as answer]
You can find the default credentials in the documentation: https://streampipes.apache.org/docs/try-installation/

I'll convert this issue to a discussion since it's probably not related to a bug.

**Reply by himanshu-pareek** [reply]
I can see the dashboard. So, it worked. Thanks @tenthe and @dominikriemer for the help.

**Comment by himanshu-pareek** [comment]
Closing the discussion as it is working as expected.
